src/main/java/com/twilio/oai/resolver/common/CodegenParameterResolver.java

```diff
@@ -0,0 +1,49 @@
+package com.twilio.oai.resolver.common;
+
+import com.twilio.oai.common.LanguageDataType;
+import com.twilio.oai.resolver.Resolver;
+import org.openapitools.codegen.CodegenParameter;
+
+import java.util.List;
+import java.util.Map;
+
+public class CodegenParameterResolver implements Resolver<CodegenParameter> {
+
+    private final Map<String, Map<String, Object>> conventionMap;
+    private List<? extends LanguageDataType> languageDataTypes;
+    private CodegenParameterDataTypeResolver codegenParameterDataTypeResolver = new CodegenParameterDataTypeResolver();
+    private CodegenParameterContainerDataTypeResolver codegenParameterContainerDataTypeResolver;
+
+    public CodegenParameterResolver(Map<String, Map<String, Object>> conventionMap,
+                                    List<? extends LanguageDataType> languageDataTypes) {
+        this.conventionMap = conventionMap;
+        this.languageDataTypes = languageDataTypes;
+        codegenParameterContainerDataTypeResolver = new CodegenParameterContainerDataTypeResolver(codegenParameterDataTypeResolver,
+                languageDataTypes);
+    }
+
+    public CodegenParameter resolve(CodegenParameter parameter) {
+        if (parameter == null) {
+            return null;
+        }
+
+        codegenParameterDataTypeResolver.setConventionMap(conventionMap);
+
+        if (isContainerType(parameter)) {
+            codegenParameterContainerDataTypeResolver.resolve(parameter);
+        } else {
+            codegenParameterDataTypeResolver.resolve(parameter);
+        }
+
+        return parameter;
+    }
+
+    private boolean isContainerType(CodegenParameter parameter) {
```

Comment:
CodegenParameter has an [isContainer](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java#L117) property that can likely be used in place of this

Reply:
Good find. Refactored to use the `isContainer` property